app/routes/titiler/algorithms/tree_cover_loss.py

```diff
@@ -0,0 +1,156 @@
+
+        # Stack RGB and alpha channels
+        data = np.vstack([rgb, alpha[np.newaxis, ...]]).astype(self.output_dtype)
+        data = np.ma.MaskedArray(data, mask=False)
+    
+        return ImageData(data, assets=img.assets, crs=img.crs, bounds=img.bounds)
+
+    def create_mask(self):
+        mask = ~self.no_data
+
+        # TCL goes from 2001 to 2023 by default, only filter if start_year or end_year is specified
+        if self.start_year != 2001:
+            start_mask = self.tree_cover_loss_data >= (self.start_year - 2000)
+            mask &= start_mask
+
+        if self.end_year != 2023:
+            end_mask = self.tree_cover_loss_data <= (self.end_year - 2000)
+            mask &= end_mask
+
+        # Threshold by TCD if specified
+        if self.tree_cover_density_data is not None:
+            if self.tree_cover_density_threshold is not None:
+                density_mask = self.tree_cover_density_data.array[0, :, :] >= self.tree_cover_density_threshold
+                mask &= density_mask
+
+        return mask
+
+    def create_encoded_rgb(self):
+        # Red = intensity
+        r = np.clip(self.intensity, 0, 255).astype("uint8")
+
+        # Green = 0 
+        g = np.zeros_like(r, dtype="uint8")
+
+        # Blue = year of loss (1–24)
+        b = np.where(self.mask, self.tree_cover_loss_data, 0).astype("uint8")
+
+        return np.stack([r, g, b], axis=0)
+        
+    def create_encoded_alpha(self):
+        # Alpha = 255 where intensity > 0 and mask is True
+        alpha = np.where((self.intensity > 0) & self.mask, 255, 0).astype("uint8")
+
+        return alpha
+    
+    def create_true_color_rgb(self):    
+        # Scale intensity based on zoom level
+        scale_pow = self.scale_intensity(self.zoom)
+        scaled_intensity = scale_pow(self.intensity).astype("uint8")
+
+        # Red = 228
+        r = np.full(self.tree_cover_loss_data.shape, 228, dtype="float32")
+        
+        # Green = 102 by default, but scales down with zoom level and intensity
+        g = (
+            np.ones(self.tree_cover_loss_data.shape, dtype="float32") * 102
+            + (72 - self.zoom) 
+            - (scaled_intensity * (3 / max(self.zoom, 1)))
+        )
+
+        # Blue = 153 by default, but scales down with zoom level and intensity
+        b = (
+            np.ones(self.tree_cover_loss_data.shape, dtype="float32") * 153
+            + (33 - self.zoom) 
+            - (self.intensity / max(self.zoom, 1))
+        )
+
+        # Clip values to [0, 255] and convert to uint8
+        # Adapted from bugfix: https://gfw.atlassian.net/browse/GTC-2916 
+        g = np.clip(g, 0, 255).astype("uint8")
+        b = np.clip(b, 0, 255).astype("uint8")
+
+        return np.stack([r.astype("uint8"), g, b], axis=0)
+
+    def create_true_color_alpha(self):
+        scale_pow = self.scale_intensity(self.zoom)
+        scaled_intensity = scale_pow(self.intensity).astype("uint8")
```

Comment:
To not introduce avoidable latency at high zoom level, I suggest doing this only for the zoom levels that need it, 
Also, the scale function doesn't do anything for zoom < 11 (ln 138), so you can adjust this to match that (if 11 is correct).